File: TerraObserver/TO.Abstractions/Planets/ICelestialMotion.cs
using Godot.Abstractions.Bases;

namespace TO.Abstractions.Planets;

/// Copyright (C) 2025 Zhu Xiaohe(aka ZeromaXHe)
/// Author: Zhu XH (ZeromaXHe)
/// Date: 2025-06-05 19:50:05
public interface ICelestialMotion : INode3D
{
    bool PlanetRevolution { get; set; }
    bool PlanetRotation { get; set; }
    bool SatelliteRevolution { get; set; }
    bool SatelliteRotation { get; set; }
}

File: TerraObserver/TO.Domains/Types/Planets/ICelestialMotion.fs
namespace TO.Domains.Types.Planets

open Godot
open TO.Domains.Types.Godots

/// Copyright (C) 2025 Zhu Xiaohe(aka ZeromaXHe)
/// Author: Zhu XH (ZeromaXHe)
/// Date: 2025-06-29 14:10:29
[<Interface>]
type ICelestialMotion =
    inherit INode3D
    // =====【Export】=====
    abstract Moon: Node3D
    abstract PlanetRevolution: bool with get, set
    abstract PlanetRotation: bool with get, set
    abstract SatelliteRevolution: bool with get, set
    abstract SatelliteRotation: bool with get, set
    abstract SatelliteRadiusRatio: float32
    abstract SatelliteDistRatio: float32
    // =====【on-ready】=====
    abstract LunarDist: Node3D
